colab/colab_utils.py

```diff
@@ -0,0 +1,1019 @@
+
+      dataset_id = int(data['datasets'][0].split('/')[-1])
+      dataset_name = data['displayName']
+      labelingjob_id= int(data['name'].split('/')[-1])
+      labelingjob_instruction= data['instructionUri']
+
+      print(f'\nLabeling dataset {dataset_name} created, with ID {dataset_id}')
+      print(f'\nData Labeling job {labelingjob_id} created')
+
+      labeling_job = LabelingJob(f'{create_label_task_args["cloud_location"]}-aiplatform.googleapis.com',
+                                 create_label_task_args["cloud_project"],
+                                 create_label_task_args["cloud_location"], labelingjob_id)
+      
+      print('Instruction URL: {}'.format(labelingjob_instruction.replace('gs://','https://storage.cloud.google.com/')))
+      print(f'Worker URL: {labeling_job.get_worker_url()}')
+      print(f'Manager URL: {labeling_job.get_manager_url()}')
+      print(f'Detailed monitoring page: https://console.cloud.google.com/vertex-ai/locations/{create_label_task_args["cloud_location"]}/labeling-tasks/{labelingjob_id}?project={create_label_task_args["cloud_project"]}')
+
+      break
+    elif i == 2:
+      print('An unexpected error occurred. Output of command was:')
+      print(child.before.decode())
+      child.close()
+      raise Exception('Job terminated unexpectedly.')
+    else:
+      child.close()
+      raise Exception('Job timed out. Full output:\n' + output.decode())
+
+  child.close()
+  return dataset_id,dataset_name,labelingjob_id,labelingjob_instruction
+
+## CLASS DEFINITION DATASETJOB
+
+def create_labeled_dataset(create_labeled_dataset_args,path_dir_args):
+  
+  child = launch_pexpect_process(
+      'create_labeled_dataset.py',
+      create_labeled_dataset_args, 
+      path_dir_args,True)
+
+  print('Creating labeled datasets...')
+  child.expect(pexpect.EOF, timeout=None)
+  child.close()
+  if child.exitstatus != 0:
+    print('An unexpected error occurred. Output of command was:')
+    print(child.before.decode())
+  else:
+    print('Labeled dataset created.')
+
+## CLASS DEFINTION IPYPLOT
+
+def concat_caption_pilimage(image_before, image_after):
+  img_before=caption_pilformat(image_before, "before")
+  img_after=caption_pilformat(image_after, "after")
+
+  w, h=img_before.size
+
+  img_concat = Image.new('RGB', (2*w, h), "white")
+  img_concat.paste(img_before, (0, 0))
+  img_concat.paste(img_after, (w, 0))
+
+  return img_concat
+
+def caption_pilformat(img_data, caption):
+  base64_encoded = base64.b64encode(img_data)
+  im_bytes = base64.b64decode(base64_encoded)
+  byte_encoded=io.BytesIO(im_bytes)
+
+  img=Image.open(byte_encoded)
+  wd, hg =img.size
+
+  img_ = Image.new('RGB', (wd+int(wd/10), hg+int(hg/5)), "white")
+  img_.paste(img, (int(wd/20),int(hg/5)))
+
+  wd, hg =img_.size
+  img_cap = ImageDraw.Draw(img_)
+  _,_,w, h = img_cap.textbbox((0, 0),caption)
+  img_cap.text(((wd-w)/2,0), caption, fill=(0, 0, 0))
+
+  return img_
+
+def ipyplot_tfrecord(path, max_examples=None):
```

Comment:
Can you rename this to something more descriptive, like "visualize_examples"? "ipyplot" is an implementation detail so generally we don't want to include that in the function name.

Reply:
change commit 6d5256b